### PR TITLE
Publish the cl_img_matrix_multiply extension specification.

### PR DESCRIPTION
**Pull request by tomasz-platek**
Publish the cl_img_matrix_multiply extension specification.

### CODE REVIEW

**Comment by SunSerega** [comment]
The last commit prob wasn't intended for this PR, since you also sent `cl_img_bitwise_ops` in #1200.

**Comment by tomasz-platek** [comment]
> The last commit prob wasn't intended for this PR, since you also sent `cl_img_bitwise_ops` in #1200.
Yes, that's right, already reverted, thank you.

File: extensions/cl_img_matrix_multiply.asciidoc [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+Copyright (c) 2024 Imagination Technologies Ltd. All Rights Reserved.
+
+== Status
+
+Final Draft
+
+== Version
+
+Built On: {docdate} +
+Version: Major.Minor.Patch
```
Comment (bashbaug) [inline extensions/cl_img_matrix_multiply.asciidoc:36]: Consider listing the extension version here.

```suggestion
Version: 1.0.0
```

File: extensions/cl_img_matrix_multiply.asciidoc [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+[source]
+----
+float4 a = (float4) (1.0f, 1.0f, 1.0f, 1.0f);
+__local float8 b;
+b = (float8) (0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f);
+
+float2 acc = (float2) (1.0f, 1.0f);
+float2 res = img_dot_interleaved_acc(a, &b, acc);
+
+printf("res = [ %f %f ]\n", res.s0, res.s1);
```
Comment (bashbaug) [inline extensions/cl_img_matrix_multiply.asciidoc:257]: It might be helpful to include what gets printed in these cases.

**Review by bashbaug — APPROVED** [review]

File: extensions/cl_img_matrix_multiply.asciidoc (hunk 2 above)
Comment (tomasz-platek) [reply extensions/cl_img_matrix_multiply.asciidoc:257]: Sounds like a good idea, I've added the results.

**Pull request merged**